pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -906,3 +915,50 @@ type NomadSDConfig struct {
 	// +optional
 	EnableHTTP2 *bool `json:"enableHTTP2,omitempty"`
 }
+type DockerswarmSDConfig struct {
+	// Address of the docker daemon
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Host string `json:"host"`
+	// ProxyConfig allows customizing the proxy behaviour for this scrape config.
+	// +optional
+	v1.ProxyConfig `json:",inline"`
+	// TLS configuration to use on every scrape request
+	// +optional
+	TLSConfig *v1.SafeTLSConfig `json:"tlsConfig,omitempty"`
+	// Role of the targets to retrieve. Must be `services`, `tasks`, or `nodes`.
+	// +kubebuilder:validation:Enum=services;tasks;nodes
```

Comment:
To comply with kube API conventions. We'll need to lower-case when generating the Prometheus config.
```suggestion
	// +kubebuilder:validation:Enum=Services;Tasks;Nodes
```